buildAndReleaseTask/index.ts

```diff
@@ -1,100 +1,205 @@
-    if(os == "Windows_NT") {
-        return "C:\\Program Files\\appknox.exe";
-    }
-    if(os == "Darwin" || os == "Linux") {
-        return "/usr/local/bin/appknox";
+
+/**
+ * Gets proxy url set via ENV, fallbacks to agent proxy
+ * @returns proxy url
+ */
+function getProxyURL(): string {
+    const envProxy = (
+        process.env.HTTPS_PROXY ||
+        process.env.https_proxy ||
+        process.env.HTTP_PROXY ||
+        process.env.http_proxy
+    );
+
+    if (envProxy && !isValidURL(envProxy)) {
+        throw Error(`Invalid proxy url in environment: ${envProxy}`);
     }
-    throw Error("unsupported os " + os);
+
+    const agentProxyConfig = tl.getHttpProxyConfiguration();
+    const agentProxy = (agentProxyConfig != null) ? agentProxyConfig.proxyUrl : "";
+
+    return envProxy || agentProxy;
 }
 
-async function copyAppknox() {
-    const appknoxPath = getAppknoxPath();
-    const binPath = path.join(__dirname, 'binaries');
-    const winBin = path.join(binPath, winBinary)
-    const darwinBin = path.join(binPath, darwinBinary);
-    const linuxBin = path.join(binPath, linuxBinary);
-    if(fs.existsSync(appknoxPath)) {
-        return;
+/**
+ * Determines whether the URL is valid
+ * @param url
+ * @returns true if valid
+ */
+function isValidURL(url: string): boolean {
+    if (!url) {
+        return false;
     }
-    await downloadBinary();
-    if(os == "Windows_NT") {
-        return tl.cp(winBin, appknoxPath);
+    return isUrlHttp(url);
+}
+
+/**
+ * Gets appknox binary download url
+ * @param os
+ * @returns url
+ */
+function getAppknoxDownloadURL(os: string): string {
+    if (!(os in supportedOS)) {
+        throw Error(`Unsupported os ${os}`);
     }
-    if(os == "Darwin") {
-        tl.cp(darwinBin, appknoxPath, "-f");
-        return fs.chmodSync(appknoxPath, "755");
+    const binaryVersion = pkg.binary;
+    const binaryName = supportedOS[os].name;
+    return `https://github.com/appknox/appknox-go/releases/download/${binaryVersion}/${binaryName}`;
+}
+
+/**
+ * Downloads file to the specified destination
+ * @param url
+ * @param proxy
+ * @param dest file
+ */
+async function downloadFile(url: string, proxy: string, dest: string): Promise<any> {
+    const opts = {
+        agent: proxy ? new ProxyAgent(proxy) : undefined,
+        follow: 5,
+        output: dest,
+    };
+    return new Promise((resolve: (value?: unknown) => void, reject: (reason?: any) => void) =>
+        needle.get(url, opts, function(err: any, resp: http.IncomingMessage, body: string) {
+            if (err) {
+                return reject(err);
+            }
+            if (resp.statusCode !== 200) {
+                return reject(new Error(`Error code ${resp.statusCode}: ${resp.statusMessage}`));
+            }
+            tl.debug(`File downloaded: ${dest}`);
+            return resolve(resp);
+        })
+    ).catch(function(err: any) {
+        tl.debug(`Error downloading file: ${err}`);
```

Comment:
this should be error no debug

Reply:
Only available methods are `warning`, `error` & `debug`